lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/data/config_pair.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBarContainer extends StatefulWidget {
+  final MaterialApp child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) =>
+      context.dependOnInheritedWidgetOfExactType<_Updater>()!.data.updateBar();
+
+  /// This method used internally.
+  /// Will initiate walk through the children widgets,
+  /// find the latest instance of [NavBarConfig] and if it is not [null],
+  /// then will draw [NavBar] with it.
+  /// If not [NavBarConfig] found - [NavBar] will be not shown(removed)
+  ///
+  /// [goingBack] is a flag, that indicates that navigation flow going back.
+  /// when [true], then will ignore latest [NavBarConfig] (from screen that will be removed)
+  /// and will try to apply previous [NavBarConfig] (from screen that will be shown instead)
+  static void resetNavBar(
+    BuildContext context, {
+    required bool goingBack,
+  }) =>
+      context
+          .dependOnInheritedWidgetOfExactType<_Updater>()!
```

Comment:
❗  Here the same with the assert instead of the `!` to give users a pretty good feedback